### ISSUE TITLE
Multiple Cycle Tasks

### ISSUE THREAD

**Issue by eb2ai** [wenlzhang/obsidian-task-marker#86]
This is the feature I didn't know I needed. I use the cycle tasks so many times in a day that my hotkeys are complaining.

Having two or even three different Cycling Tasks lists would be great.

I have many checklists that I use for different purposes. For example, I use the current Cycle Tasks for... wait for it... tasks. It looks like this.  `x/->:fr `

While taking notes in meetings, I will often use any of these `g+"!w:f` to quickly capture a group comment, quote, win, or something to follow up on later. Cycling through this list also would be amazing.

Thanks for considering this enhancement.

**Comment by wenlzhang** [comment]
This shouldn't be difficult to implement. I'll take a look at this later.

Considering your feedback, I am thinking about four lists for now.
- One (main) list that can contain a complete list of statuses. This can be seen as the current implementation of cycling list.
- Three (small) lists that can contain fewer statuses than the main list but can be used for different purposes.

Considering the above settings, the option "Support cycling task status reversely" may not be added to the three small lists, as there are fewer statuses included.

What do you think about this?

**Comment by eb2ai** [comment]
I think that sounds amazing. Thanks! 

Is there a performance concern, considering the size of the lists? If not, I'm not sure making them any less than the others is really an issue. 

I'm excited to see this added to the plugin.

**Comment by wenlzhang** [comment]
> Is there a performance concern, considering the size of the lists?

The user can define what can be included in the "complete" list of statuses to cycle among. I did not test this extensively, but I hope there are no performance issues with this.

For other lists, I did not encounter performance issues either after some tests.

I just released a new version that adds the request feature. Just me know if it satisfies your needs or there are any issues.
